server/server.ts

```diff
@@ -100,6 +113,21 @@ export default async function (
 		.use(express.static(Utils.getFileFromRelativeToRoot("public"), staticOptions))
 		.use("/storage/", express.static(Config.getStoragePath(), staticOptions));
 
+	issuer = await Issuer.discover(Config.values.openid.issuerURL);
+	log.info("Discovered OpenID issuer", issuer.metadata.issuer);
+	openidClient = new issuer.Client({
+		client_id: Config.values.openid.clientID,
+		client_secret: Config.values.openid.secret,
+		redirect_uris: [Config.values.openid.baseURL],
+		response_types: ["code"],
+	});
+	var redirectUrl = openidClient.authorizationUrl({
+		scope: "openid email profile",
+		code_challenge,
+		code_challenge_method: "S256",
+	});
+	issuerURL = redirectUrl;
+
```

Comment:
I don't think that the server should be handling any specifics of the openid auth whatsoever.

Shove that into an abstraction layer with a simple UI.
You shouldn't even be resolving the issuer unless openid auth is enabled.

Reply:
>  You shouldn't even be resolving the issuer unless openid auth is enabled.

Whoops, I forgot to do that. I'll get that done.

> I don't think that the server should be handling any specifics of the openid auth whatsoever.

How do you mean? Since OpenID works by communicating securely with the server (the server has to validate the client's tokens with the OpenID server), then the server has to handle specifics and make sure that the encryption algorithms and all are consistent.

Reply:
Not servers side vs client side.
That you do the auth/basic flow server side is not the point.

What you do however is shove logic related to openid into `server.ts`
Have a look at how say the uploader or ident functionality is wired up. `server.ts` has absolutely no knowledge about those other than that it's wiring them up.

You are inlining openid specific logic.